src/ui/public/state_management/state_storage/_tests__/hashed_item_store.js

```diff
@@ -0,0 +1,285 @@
+import expect from 'expect.js';
+import sinon from 'sinon';
+
+import StubBrowserStorage from 'test_utils/stub_browser_storage';
+import HashedItemStore from '../hashed_item_store';
+
+describe('hashedItemStore', () => {
+  describe('#constructor', () => {
+    it('retrieves persisted index from sessionStorage', () => {
+      const sessionStorage = new StubBrowserStorage();
+      sinon.spy(sessionStorage, 'getItem');
+
+      const hashedItemStore = new HashedItemStore(sessionStorage);
+      sinon.assert.calledWith(sessionStorage.getItem, HashedItemStore.PERSISTED_INDEX_KEY);
+      sessionStorage.getItem.restore();
+    });
+
+    it('builds index from persisted index', () => {
+      const sessionStorage = new StubBrowserStorage();
+
+      const persistedItemIndex = {
+        a: {
+          hash: 'a',
+          touched: 1,
+        },
+        b: {
+          hash: 'b',
+          touched: 0,
+        },
+      };
+
+      sessionStorage.setItem(
+        HashedItemStore.PERSISTED_INDEX_KEY,
+        JSON.stringify(persistedItemIndex),
+      );
+
+      const hashedItemStore = new HashedItemStore(sessionStorage);
+
+      expect(hashedItemStore._items).to.eql([persistedItemIndex.b, persistedItemIndex.a]);
+
+      expect(hashedItemStore._itemIndex).to.eql({
+        [persistedItemIndex.a.hash]: persistedItemIndex.a,
+        [persistedItemIndex.b.hash]: persistedItemIndex.b,
+      });
+    });
+  });
+
+  describe('#setItem', () => {
+    describe('if the item exists in sessionStorage', () => {
```

Comment:
I haven't read the docs, but I use `describe()` to define what I'm testing, and I (mis)use `context()` to describe the properties I've given the item, or situation I'm putting it in, in order to accomplish the test.

Put another way, `describe('noun')`, `context('adjective')` or `context('scenario')`

Which would look something like this:

```js
describe('HashedItemStore', () => {
  describe('#setItem', () => {
    context('has is not in sessionStorage', () => {
      it('stores the value', () => ...)
      it('returns the stored value', () => ...)
    })
    context('hash is already in sessionStorage', () => {
      it('returns the stored value', () => ...)
    })
  })
})
```